### ISSUE TITLE
Library doesn't create nested directories in trash.

### ISSUE THREAD

**Issue by magmast** [Byron/trash-rs#45]
I'm developing safe `rm` command alternative. It's available [here](https://gitlab.com/magmast/sfrm), if you want to test the issue on the same program that I do.

The problem is that the library cannot remove nested directories on linux (Fedora 36 in my case, but I suppose it doesn't work in other distros either). It works perfectly with files or directories that contain only files, but when directory contains other directories, they're not created in trash and it makes `rename` operation fail.

You can try that using `sfrm`:

```sh
git clone https://gitlab.com/magmast/sfrm.git
cd sfrm

mkdir a
touch a/b
RUST_LOG=trace cargo run -- -r a # Here everything works and you can see that files are in trash

mkdir -p a/b
touch a/b/c
RUST_LOG=trace cargo run -- -r a # This operation fails
```

Output from the failing command:

```sh
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/sfrm -r a`
2022-03-27T21:33:18.057567Z TRACE trash: Starting canonicalize_paths    
2022-03-27T21:33:18.057622Z TRACE trash: Finished canonicalize_paths    
2022-03-27T21:33:18.057822Z DEBUG trash::platform: The home topdir is "/home"    
2022-03-27T21:33:18.057849Z DEBUG trash::platform: Deleting "/home/magmast/Projects/sfrm/a"    
2022-03-27T21:33:18.057879Z DEBUG trash::platform: The topdir of this file is "/home"    
2022-03-27T21:33:18.057891Z DEBUG trash::platform: The topdir was identical to the home topdir, so moving to the home trash.    
2022-03-27T21:33:18.057984Z DEBUG trash::platform: Successfully created "/home/magmast/.local/share/Trash/info/a.trashinfo"    
2022-03-27T21:33:18.058199Z DEBUG trash::platform: Failed moving item to the trash (this is usually OK). Os { code: 2, kind: NotFound, message: "No such file or directory" }    
Cannot remove 'a': Error during a `trash` operation: Unknown { description: "Path: '\"/home/magmast/.local/share/Trash/files/a\"'. Message: No such file or directory (os error 2)" }
```

I've managed to find what the issue is (not creating nested directories in trash) and fixed it [here](https://github.com/magmast/trash-rs/tree/fix/no-such-file-or-directory). I'll make pull request in a while, but the solution I made probably is a dirty hack and I'll would be grateful for instructions how to make the code I wrote to mach the rest of the library better.

**Comment by Byron** [comment]
Thanks for letting me know!

I will head over to the PR for further comments.

> It's available [here](https://gitlab.com/magmast/sfrm), if you want to test the issue on the same program that I do.

The link doesn't work for me (404), maybe the repo is private (I am logged-in there as well)?

**Comment by magmast** [comment]
> Thanks for letting me know!
> 
> I will head over to the PR for further comments.
> 
> > It's available [here](https://gitlab.com/magmast/sfrm), if you want to test the issue on the same program that I do.
> 
> The link doesn't work for me (404), maybe the repo is private (I am logged-in there as well)?

Yeah I forgot to make it public. It should work now.

**Comment by aalhitennf** [comment]
Got same error  

```
use std::fs::{File, DirBuilder};

fn main() {
    DirBuilder::new().recursive(true).create("remove-me/dir1").unwrap();
    DirBuilder::new().recursive(true).create("remove-me/dir2").unwrap();
    File::create("remove-me/dir1/remove-me").unwrap();
    File::create("remove-me/dir2/remove-me").unwrap();

    trash::delete("remove-me").unwrap();
    assert!(!std::path::Path::new("remove-me").exists());
}
```

Results to:  

> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Unknown { description: "Path: '\"/home/myuser/.local/share/Trash/files/remove-me\"'. Message: No such file or directory (os error 2)" }', src/main.rs:9:32

**Comment by Byron** [comment]
This is fixed in [v2.1.0](https://github.com/Byron/trash-rs/releases/tag/v2.1.0).

Thanks for providing a test-case and for the PR which provided the fix (it was missing a test and thus wasn't merged yet).
